test/form-builder/NumberField.test.tsx

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+/* eslint-disable @typescript-eslint/no-empty-function */
+
+import React from 'react';
+import { waitFor } from '@testing-library/react';
+
+import NumberField from '../../src/form-builder/NumberField';
+import { buildRenderForm, changeValue } from '../utils';
+
+const renderForm = buildRenderForm({});
+
+const getInput = (container: HTMLElement): HTMLVaNumberInputElement => {
+  const input = container.querySelector(
+    'va-number-input'
+  ) as HTMLVaNumberInputElement;
+  if (!input) throw new Error('No va-number-input found');
+  return input;
+};
+
+describe('Form Builder - NumberField', () => {
+  test('renders', () => {
+    const { container } = renderForm(
+      <NumberField name="thing" label="The Thing" />
+    );
+    const input = getInput(container);
+    expect(input.getAttribute('label')).toEqual('The Thing');
+    expect(input.getAttribute('name')).toEqual('thing');
+  });
+
+  test('renders initial value', () => {
+    const rf = buildRenderForm({ thing: 'asdf' });
+    const { container } = rf(<NumberField name="thing" label="The Thing" />);
+    const input = getInput(container);
+    expect(input.getAttribute('value')).toEqual('asdf');
```

Comment:
A non-numerical input should not be allowed.

Reply:
Nevermind, I think this is just changing the Formik related value and not the actual input field value. There's a unit test that covers this functionality in `VaNumberInput`.